File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.java
import java.util.*;

class Solution {
    public String[] solution(String myStr) {
        String[] arr = myStr.split("a|b|c");
        List<String> li = new ArrayList<String>();
        for(String str : arr) {
        	if(!str.isEmpty()) li.add(str);
        }
        if(li.size() > 0){
            return li.toArray(new String[li.size()]);    
        } else{
            return  new String[]{"EMPTY"};
        }
        
    }
}

File: 프로그래머스/0/181862. 세 개의 구분자/세 개의 구분자.cpp
#include <bits/stdc++.h>

using namespace std;

vector<string> solution(string myStr) {
    for(char &ch : myStr) {
        if(ch == 'a' || ch == 'b' || ch == 'c') ch = ' ';
    }
    
    std::vector<std::string> ans;
    std::istringstream iss(myStr);
    std::string word;
    while(iss >> word) ans.push_back(word);
    
    if(ans.size() == 0) ans.push_back("EMPTY");
    return ans;
}